### ISSUE TITLE
Pattern Assembler - Keep the pattern selector open when adding/replacing header/footer pattern

### ISSUE THREAD

**Issue by miksansegundo** [Automattic/wp-calypso#70473]
When we add a header or footer or replace any pattern, we use one click to do the action, and the UI automatically goes back to the list of selected patterns.

A possible improvement is to keep the pattern selector open when adding/replacing a pattern so users can test different patterns without going back and ford to the list of selected patterns. When ready, they can click on `Done` to go back to the list of selected patterns.

## Acceptance criteria
- Keep the pattern selector open when adding a header or footer or replacing any pattern
- When we click on a pattern, it is previewed on the right without going back to the list of selected patterns
- When ready, click on a new button `Done` or on the arrow on the title to go back to the list of selected patterns

@fushar, Is this what you mean? 
@SaxonF, what are your thoughts?

Related https://github.com/Automattic/wp-calypso/issues/70461

**Comment by fushar** [comment]
> @fushar, Is this what you mean?

Yup, exactly, thanks!

**Comment by fushar** [comment]
After actually playing around with the feature (via Calypso Live on @miksansegundo's PRs), I just realized that this inconsistency might confuse the user. Let's say that the user clicks "Add header", tries out several patterns and then clicks Done. Now, they click "Add sections". They will think that the behavior is the same: replacing, not appending, because there is no indicator/education at all.

Let's say that they appended a pattern by mistake (they actually wanted to replace the first section pattern). They will be confused, and they will be even more confused as to how to delete the pattern -- there is currently no way to do that from the inserter screen. Also, the button says "Done" where the user is not done at all.

My suggestions:

1. Add small subtitle below the `< Add sections` heading: something like `You can select multiple patterns. Click a pattern to append it to your page` (not sure about the copy though)
2. "Done" -> "Back"

Thoughts? @SaxonF @miksansegundo

**Comment by miksansegundo** [comment]
Nice idea, let's do that 👍

Also we could add a brief description of what a header, a footer and a pattern are. I think it is an opportunity to introduce the basic concepts. We have collected great ideas for the copy in the CfT.

I remember Matias mentioned about these descriptions in the call and about adding categories too in v2. I found that they are actively working on updating the categories in the editor.
